mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java

```diff
@@ -140,13 +141,17 @@ public void setPresenter(EditProfileContract.EditProfilePresenter presenter) {
 
     @OnClick(R.id.fab_edit_profile_save_changes)
     public void onSaveChangesClicked() {
+        hideKeyboard();
+        if (!Utils.isValidEmail(etEmail.getText().toString())) {
+            onUpdateEmailError(Constants.INVALID_EMAIL);
+            return;
+        }
         for (EditText input : userDetailsInputs) {
             if (isDataSaveNecessary(input)) {
                 mPresenter.updateInputById(input.getId(), input.getText().toString());
             }
         }
         hideFab();
-        hideKeyboard();
```

Comment:
Please explain why are you removing this? Am I missing out on something?
@akkywadhwa

Reply:
@shiv07tiwari I have not removed but shifted `hideKeyboard()` to close keyboard before toast appears. This is to avoid overlapping of keyboard and toast, making toast clearly visible.

Reply:
@shiv07tiwari @luckyman20 Can you please answer this?